test/unit/foreman_kubevirt_test.rb

```diff
@@ -5,7 +5,36 @@ class ForemanKubevirtTest < ActiveSupport::TestCase
     User.current = User.find_by login: 'admin'
   end
 
-  test 'the truth' do
-    assert true
+  def new_kubevirt_vcr(proto = 'http')
+    ComputeResource.new_provider(
+      :provider => "Kubevirt",
+      :name => 'kubevirt-multus',
+      :hostname => "192.168.111.13",
+      :api_port => "6443",
+      :namespace => "default",
+      :token => "kubetoken"
+    )
+  end
+
+  test "test_connection should fail if client does not support virt" do
+    client = stub()
+    client.stubs(:virt_supported?).returns(false)
+    client.stubs(:valid?).returns(true)
+    record = new_kubevirt_vcr
+    record.stubs(:client).returns(client)
+    assert_equal false, record.test_connection
+  end
+
+  test "create_vm network based" do
+    require 'kubeclient'
+    vm_args = {"cpu_cores"=>"1", "memory"=>"1073741824", "start"=>"1", :volumes_attributes=>{:_delete=>"", :storage_class=>"local-storage", :name=>"claimski1", :capacity=>"3", :bootable=>"true"}, "name"=>"norma-huish.example.com", "provision_method"=>"build", "firmware_type"=>:bios, "interfaces_attributes"=>{"0"=>{"cni_provider"=>"multus", "network"=>"ovs-foreman", "ip"=>"192.168.111.185", "mac"=>"a2:a4:b4:a8:a6:a6", "provision"=>true}}}
```

Comment:
that looks ok, but we probably have a bug somewhere with volumes_attributes that suppose to support multiple volumes, same goes for interfaces.

So I'd add a comment to add a test with multiple volumes.